### PR TITLE
Hardware debugging

### PR DESCRIPTION
**Pull request by aninternetof**
Sorry for the big PR. These are all things that came up while I was getting the development branch running on the real Rover hardware. Here's summary:

* Fixed little typos in the API and WS protocol constants.
* I was using the wrong GrovePi motor driver library. This switches to the right one.
* Changed the systemd service to start up using Docker.
* I don't think we need the *setup.sh* or *start.sh* scripts at all anymore, now that we're committed to Docker.
* Added a check for a `DEVELOPMENT` env variable to specify when you're not running on the RaspPi, instead of trying to auto-detect by catching broad exceptions.
* Updated the old README screenshot and added a fun GIF!

### CODE REVIEW

**Comment by aninternetof** [comment]
Coverage is 100%.

File: rovercode/drivers/grove_motors.py [hunk 1]
```diff
@@ -1,43 +1,66 @@
...
-        direction_bits = \
-            (self.directions[self.PORT_A] << 2) + \
-            (self.directions[self.PORT_B])
-        self.interface.DirectionSet(direction_bits)
+    def _set_speed(self, motor, speed, direction):
+        if direction not in self.DIRECTIONS:
+            LOGGER.error("Invalid direction %s", direction)
+            return
+        if speed > 100:
+            LOGGER.warning("Capping speed %s to 100", speed)
```
Comment (cabarnes) [inline rovercode/drivers/grove_motors.py]: You need to actually change `speed` here to cap it

File: rovercode/drivers/grove_motors.py [hunk 2]
```diff
@@ -1,43 +1,66 @@
...
+        if speed < 0:
+            LOGGER.warning("Applying floor to set speed %s to 0", speed)
```
Comment (cabarnes) [inline rovercode/drivers/grove_motors.py]: Here too

File: rovercode/drivers/grove_motors.py [hunk 3]
```diff
@@ -1,43 +1,66 @@
...
+        self._set_speed('right', speed, direction)
 
-        direction_bits = \
-            (self.directions[self.PORT_A] << 2) + \
-            (self.directions[self.PORT_B])
-        self.interface.DirectionSet(direction_bits)
+    def _set_speed(self, motor, speed, direction):
+        if direction not in self.DIRECTIONS:
+            LOGGER.error("Invalid direction %s", direction)
+            return
```
Comment (cabarnes) [inline rovercode/drivers/grove_motors.py]: I think you should probably check that `motor` is valid here as well

File: rovercode/drivers/grove_motors.py (hunk 1 above)
Comment (aninternetof) [reply rovercode/drivers/grove_motors.py]: Done!

File: rovercode/drivers/grove_motors.py (hunk 2 above)
Comment (aninternetof) [reply rovercode/drivers/grove_motors.py]: Done!

File: rovercode/drivers/grove_motors.py (hunk 3 above)
Comment (aninternetof) [reply rovercode/drivers/grove_motors.py]: Done!

**Comment by aninternetof** [comment]
Removing all those `apt-get install`s sure made our Travis build go faster.

**Pull request merged**